regparser/tree/reg_text.py

```diff
@@ -84,6 +85,95 @@ def build_subpart(text, part):
         "", [], label, subpart_title, node_type=struct.Node.SUBPART)
 
 
+def subjgrp_label(starting_text, letter_list):
+    words = starting_text.split(" ")
+    candidate_text = ""
+    suffixes = [""] + [_ for _ in string.lowercase[1:]]
+    if len(words) == 1:
+        """
+        E.g. if the word is "Penalties" the progression is:
+
+        Pe
+        Pe.
+        Pen
+        Pen.
+        Pena
+        Pena.
+        <etc.>
+        Penalties.
+        Penalties-b.
+        Penalties-c.
+        <etc.>
+        """
+        word = words[0]
+        terminators = ("", ".")
+        suffix_pos, terminator_pos = 0, 0
```

Comment:
It seems that the terminators have semantic meaning rather than positional meaning -- put a different way, this uses the terminators as an `Enum` rather than a list. That's cool, but then we probably shouldn't have the list. Something like:

```python
if not terminator:
    terminator = "."
    continue
else:
    terminator = ""
```